src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1569,15 +1574,109 @@ bool DataWriterImpl::deadline_timer_reschedule()
     return true;
 }
 
+/**
+ * This function tears down any existing deadline timer and creates a new one
+ * configured for the current deadline period. If the period is 0 or infinite,
+ * the timer is created to fire only once to log a warning and then cancel itself.
+ */
+void DataWriterImpl::configure_deadline_timer_locked_()
+{
+    if (deadline_timer_ != nullptr)
+    {
+        deadline_timer_->cancel_timer();
+        delete deadline_timer_;
+        deadline_timer_ = nullptr;
+    }
+
+    // If we had previously hit the max, keep timer disabled until QoS changes reset the count.
+    if (deadline_missed_status_.total_count >= std::numeric_limits<int32_t>::max())
+    {
+        return; // no new timer
+    }
```

Comment:
I believe this condition will never be given. If that's the case, I would this block to make the code more clear.